components/ProfilePage/ProfilePage.tsx

```diff
@@ -39,13 +40,19 @@ export function ProfilePage(profileprops: {
     profile?.role === "pendingUpgrade" ||
     false
 
+  const [isProfilePublic, setIsProfilePublic] = useState<boolean>(false)
```

Comment:
Not sure this state is necessary if `profile?.public` exists?

Reply:
profile?.public will not cause the <Banner> component to properly rerender to show current "publicity" when the "Make Public/Private" button is used.  The leads to the <Banner> displaying incorrect/unsynced information.

a state variable was introduced to cause the <Banner> to properly rerender when the "Make Public/Private" button updates the profile on the backend.